### DISCUSSION TITLE
What is "it", which is the best choice if targeting multiple platforms?

### DISCUSSION THREAD

**Discussion by JDLH** [linkedin/shiv#218] [Q&A]
What does the word "it" refer to, in the sentence:

> On Windows, the Python launcher `py` knows how to handle shebangs using `env`, so it’s overall the best choice if you target multiple platforms with a pure Python zipapp.

(Source: the documentation's [_Choosing a Python Interpreter Path_ section](https://shiv.readthedocs.io/en/latest/cli-reference.html#choosing-a-python-interpreter-path).) 

This use of the word "it" seems unclear to me. By straightforward meaning, "it" could refer to Windows, or to the Python launcher named "py", or to "env". But maybe "it" refers to "the extension `.py` on the output file"?

Should I interpret the sentence as, 

> If you are targeting multiple platforms with an app made by `shiv`, it is overall the best choice to name the output file with an extension ".py". This is because on Windows, the extension ".py" activates the Python launcher `py.exe`, which will interpret the shebang in the output file. On other platforms, the extension is disregarded and the shebang runs the app correctly.

(I'm leaving out the term zipapp, because I suspect that someone packaging an app with "shiv" does not need to know that the resulting file is actually a zipapp.)

**Comment by lorencarvalho** [comment]
Hi @JDLH

Sorry for the delayed response, I actually didn't notice that this project had a `discussion` tab until recently. This sentence requires the context of the prior paragraph, but could certainly be more clear/obvious in it's wording. Perhaps something like:

> On Windows, the Python launcher `py` knows how to handle shebangs using env, so using `/usr/bin/env python3` is overall the best choice if you target multiple platforms with a pure Python zipapp.

The only nitpick I would offer your interpretation is the comment about other platforms disregarding the extension, only because that's not _necessarily_ the case.
